pkg/utils/integrations.go

```diff
@@ -0,0 +1,116 @@
+package utils
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/apono-io/apono-cli/pkg/aponoapi"
+	"github.com/apono-io/apono-cli/pkg/clientapi"
+)
+
+const (
+	integrationNameSeparator = "/"
+)
+
+func ListIntegrations(ctx context.Context, client *aponoapi.AponoClient) ([]clientapi.IntegrationClientModel, error) {
+	return GetAllPages(ctx, client, func(ctx context.Context, client *aponoapi.AponoClient, skip int32) ([]clientapi.IntegrationClientModel, *clientapi.PaginationClientInfoModel, error) {
+		resp, _, err := client.ClientAPI.InventoryAPI.ListIntegration(ctx).
+			Skip(skip).
+			Execute()
+		if err != nil {
+			return nil, nil, err
+		}
+
+		return resp.Data, &resp.Pagination, nil
+	})
+}
+
+func ListResourceTypes(ctx context.Context, client *aponoapi.AponoClient, integrationID string) ([]clientapi.ResourceTypeClientModel, error) {
+	return GetAllPages(ctx, client, func(ctx context.Context, client *aponoapi.AponoClient, skip int32) ([]clientapi.ResourceTypeClientModel, *clientapi.PaginationClientInfoModel, error) {
+		resp, _, err := client.ClientAPI.InventoryAPI.ListResourceTypes(ctx).
+			IntegrationId(integrationID).
+			Skip(skip).
+			Execute()
+		if err != nil {
+			return nil, nil, err
+		}
+
+		return resp.Data, &resp.Pagination, nil
+	})
+}
+
+func ListResources(ctx context.Context, client *aponoapi.AponoClient, integrationID string, resourceType string) ([]clientapi.ResourceClientModel, error) {
+	return GetAllPages(ctx, client, func(ctx context.Context, client *aponoapi.AponoClient, skip int32) ([]clientapi.ResourceClientModel, *clientapi.PaginationClientInfoModel, error) {
+		resp, _, err := client.ClientAPI.InventoryAPI.ListResources(ctx).
+			IntegrationId(integrationID).
+			ResourceTypeId(resourceType).
+			Skip(skip).
+			Execute()
+		if err != nil {
+			return nil, nil, err
+		}
+
+		return resp.Data, &resp.Pagination, nil
+	})
+}
+
+func ListPermissions(ctx context.Context, client *aponoapi.AponoClient, integrationID string, resourceType string) ([]clientapi.PermissionClientModel, error) {
+	return GetAllPages(ctx, client, func(ctx context.Context, client *aponoapi.AponoClient, skip int32) ([]clientapi.PermissionClientModel, *clientapi.PaginationClientInfoModel, error) {
+		resp, _, err := client.ClientAPI.InventoryAPI.ListPermissions(ctx).
+			IntegrationId(integrationID).
+			ResourceTypeId(resourceType).
+			Skip(skip).
+			Execute()
+		if err != nil {
+			return nil, nil, err
+		}
+
+		return resp.Data, &resp.Pagination, nil
+	})
+}
+
+func GetIntegrationByIDOrTypePlusName(ctx context.Context, client *aponoapi.AponoClient, integrationIDOrName string) (*clientapi.IntegrationClientModel, error) {
+	integrations, err := ListIntegrations(ctx, client)
+	if err != nil {
+		return nil, err
+	}
+
+	var integration *clientapi.IntegrationClientModel
+	if strings.Contains(integrationIDOrName, integrationNameSeparator) {
+		parts := strings.Split(integrationIDOrName, integrationNameSeparator)
+		if len(parts) != 2 {
```

Comment:
theoretically there can be a `/` in the integration name, no?

Reply:
UUID should never contain /, only digits, letters and "-"